components/SnippetDetails.js

```diff
@@ -0,0 +1,36 @@
+import Link from "next/link";
+import { useRouter } from "next/router";
+import useSWR from "swr";
+
+function SnippetDetails() {
+  const router = useRouter();
+
+  const { id } = router.query;
+
+  const { data, isLoading, error } = useSWR(`/api/snippets/${id}`);
+
+  if (error) return <div>failed to load Details View 🥺</div>;
+  if (isLoading) {
+    return <h1>Loading Details View 🤓</h1>;
```

Comment:
better to use a div, since you already have an h1 form your Header component (and there should always only be one h1 displayed)